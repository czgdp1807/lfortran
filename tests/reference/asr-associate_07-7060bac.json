{
    "basename": "asr-associate_07-7060bac",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/associate_07.f90",
    "infile_hash": "fe0eb1af053495fc080e49f70bf829f7aee3ac2f87c5c64b552bdffe",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-associate_07-7060bac.stdout",
    "stdout_hash": "dae18731c48d82cafe563c6e8e4e28f441b49e8da7fbd7c398203ace",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}