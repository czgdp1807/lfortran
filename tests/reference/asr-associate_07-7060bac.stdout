(TranslationUnit
    (SymbolTable
        1
        {
            associate_07:
                (Program
                    (SymbolTable
                        2
                        {
                            p1:
                                (Variable
                                    2
                                    p1
                                    []
                                    Local
                                    (NullPointerConstant
                                        (Pointer
                                            (Integer 4 [])
                                        )
                                    )
                                    ()
                                    Save
                                    (Pointer
                                        (Integer 4 [])
                                    )
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            p2:
                                (Variable
                                    2
                                    p2
                                    []
                                    Local
                                    (NullPointerConstant
                                        (Pointer
                                            (Real 8 [(()
                                            ())
                                            (()
                                            ())])
                                        )
                                    )
                                    ()
                                    Save
                                    (Pointer
                                        (Real 8 [(()
                                        ())
                                        (()
                                        ())])
                                    )
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            p3:
                                (Variable
                                    2
                                    p3
                                    []
                                    Local
                                    (NullPointerConstant
                                        (Pointer
                                            (Complex 4 [(()
                                            ())
                                            (()
                                            ())])
                                        )
                                    )
                                    ()
                                    Save
                                    (Pointer
                                        (Complex 4 [(()
                                        ())
                                        (()
                                        ())])
                                    )
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            t1:
                                (Variable
                                    2
                                    t1
                                    []
                                    Local
                                    (IntegerConstant 2 (Integer 4 []))
                                    ()
                                    Save
                                    (Integer 4 [])
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            t2:
                                (Variable
                                    2
                                    t2
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8 [((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 2 (Integer 4 [])))
                                    ((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 3 (Integer 4 [])))])
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            t3:
                                (Variable
                                    2
                                    t3
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Complex 4 [((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 2 (Integer 4 [])))
                                    ((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 4 (Integer 4 [])))])
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    associate_07
                    []
                    [(=>
                        (Var 2 p1)
                        (Var 2 t1)
                    )
                    (=>
                        (Var 2 p2)
                        (Var 2 t2)
                    )
                    (If
                        (LogicalNot
                            (Associated
                                (Var 2 p1)
                                ()
                                (Logical 4 [])
                                ()
                            )
                            (Logical 4 [])
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (LogicalNot
                            (Associated
                                (Var 2 p2)
                                ()
                                (Logical 4 [])
                                ()
                            )
                            (Logical 4 [])
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (LogicalNot
                            (Associated
                                (Var 2 p1)
                                (Var 2 t1)
                                (Logical 4 [])
                                ()
                            )
                            (Logical 4 [])
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (LogicalNot
                            (Associated
                                (Var 2 p2)
                                (Var 2 t2)
                                (Logical 4 [])
                                ()
                            )
                            (Logical 4 [])
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (Associated
                            (Var 2 p3)
                            ()
                            (Logical 4 [])
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )
                    (If
                        (Associated
                            (Var 2 p3)
                            (Var 2 t3)
                            (Logical 4 [])
                            ()
                        )
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                )
        })
    []
)
